File: sdk/typescript/examples/01-basic.ts
/**
 * Basic Usage - Push, Pull, Ack
 */
import { FlashQ } from '../src';

const client = new FlashQ();

// Push a job
const job = await client.push('emails', {
  to: 'user@example.com',
  subject: 'Hello',
});
console.log('Pushed:', job.id);

// Pull and process
const pulled = await client.pull('emails');
if (pulled) {
  console.log('Processing:', pulled.data);
  await client.ack(pulled.id);
  console.log('Done!');
}

await client.close();

File: sdk/python/examples/01_basic.py
"""
Example 01: Basic Usage

Demonstrates basic push, pull, and ack operations.
"""

import asyncio
from flashq import FlashQ


async def main():
    # Create client and connect
    async with FlashQ() as client:
        # Push a job
        job_id = await client.push("emails", {
            "to": "user@example.com",
            "subject": "Hello",
            "body": "Welcome to flashQ!",
        })
        print(f"Pushed job: {job_id}")

        # Pull the job
        job = await client.pull("emails")
        if job:
            print(f"Pulled job: {job.id}")
            print(f"Data: {job.data}")

            # Acknowledge completion
            await client.ack(job.id, {"sent": True})
            print("Job completed!")


if __name__ == "__main__":
    asyncio.run(main())
